equinix/resource_fabric_connection.go

```diff
@@ -0,0 +1,265 @@
+		"change_log":      changeLogToTerra(conn.ChangeLog),
+		"redundancy":      redundancyToTerra(conn.Redundancy),
+		"notifications":   notificationToTerra(conn.Notifications),
+		"account":         accountToTerra(conn.Account),
+		"a_side":          connectionSideToTerra(conn.ASide),
+		"z_side":          connectionSideToTerra(conn.ZSide),
+		"additional_info": additionalInfoToTerra(conn.AdditionalInfo),
+		"project":         projectToTerra(conn.Project),
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return diags
+}
+
+func resourceFabricConnectionUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf(" inside fabric connection update resource ")
+	client := meta.(*Config).fabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*Config).FabricAuthToken)
+	uuid := d.Id()
+	if uuid == "" {
+		return diag.Errorf("No connection found for the value uuid %v ", uuid)
+	}
+	dbConn := v4.Connection{}
+	var err error
+	dbConn, err = waitUntilConnectionIsActive(uuid, meta, ctx)
+	if err != nil {
+		return diag.Errorf(" Either timed out or errored out while fetching connection for uuid %s and error %v", uuid, err)
+	}
+	update, err := getUpdateRequest(dbConn, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updates := []v4.ConnectionChangeOperation{update}
+	_, res, err := client.ConnectionsApi.UpdateConnectionByUuid(ctx, updates, uuid)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf(" Error response for the connection update, response %v, error %v", res, err))
+	}
+	updatedConn := v4.Connection{}
+	updatedConn, err = waitForConnectionUpdateCompletion(uuid, meta, ctx)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf(" Errored while waiting for successful connection update, response %v, error %v", res, err))
+	}
+
+	d.SetId(updatedConn.Uuid)
+	return setFabricMap(d, updatedConn)
+}
+
+func waitForConnectionUpdateCompletion(uuid string, meta interface{}, ctx context.Context) (v4.Connection, error) {
+	log.Println("Waiting for connection update to complete")
+	stateConf := &resource.StateChangeConf{
+		Target: []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			client := meta.(*Config).fabricClient
+			dbConn, _, err := client.ConnectionsApi.GetConnectionByUuid(ctx, uuid, nil)
+			if err != nil {
+				return "", "", err
+			}
+			updatableState := ""
+			if "COMPLETED" == dbConn.Change.Status {
+				updatableState = dbConn.Change.Status
+			}
+			return dbConn, updatableState, nil
+		},
+		Timeout:    2 * time.Minute,
+		Delay:      30 * time.Second,
+		MinTimeout: 30 * time.Second,
+	}
+
+	inter, err := stateConf.WaitForStateContext(ctx)
+	dbConn := v4.Connection{}
+
+	if err == nil {
+		dbConn = inter.(v4.Connection)
+	}
+	return dbConn, err
+}
+
+func waitUntilConnectionIsActive(uuid string, meta interface{}, ctx context.Context) (v4.Connection, error) {
+	log.Println("Waiting for connection to be in active state ")
```

Comment:
the uuid may be helpful in this log message

Reply:
taken care